Automattic/wp-calypso
# My Home: Add renew expiring plan and reactivate expired plan tasks

#### Changes proposed in this Pull Request

Add new tasks for renewing a plan expiring in the next 30 days and for reactivating a plan that has expired.

User | Expiring | Expired
--- | --- | ---
Owner | <img width="975" alt="Screen Shot 2020-11-12 at 16 13 34" src="https://user-images.githubusercontent.com/1233880/98959477-9af7e900-2503-11eb-9abb-bfed49764e81.png"> | <img width="972" alt="Screen Shot 2020-11-12 at 16 18 40" src="https://user-images.githubusercontent.com/1233880/98959512-a4815100-2503-11eb-9a1e-06ba7169dc28.png">
Non-owner | <img width="1059" alt="Screen Shot 2020-11-13 at 14 06 19" src="https://user-images.githubusercontent.com/1233880/99075490-de149380-25b9-11eb-856d-4e30e25b2c90.png"> | <img width="1068" alt="Screen Shot 2020-11-13 at 14 07 48" src="https://user-images.githubusercontent.com/1233880/99075514-e8369200-25b9-11eb-8066-c3e63896ed88.png">

You may note there are a few differences from the mockups designed in https://github.com/Automattic/wp-calypso/issues/43496:
- Expiry date is displayed in a relative format (e.g. "in 3 days, 2 days ago, today") rather than in an absolute format (e.g. November 20, 2020). This is intended in order to be aligned with how expiry dates are displayed in Plan > Billing (`/purchases/subscriptions/:site`): <img width="963" alt="Screen Shot 2020-11-12 at 16 39 17" src="https://user-images.githubusercontent.com/1233880/98961253-a9470480-2505-11eb-9f12-9c8447e74500.png">
- The "Action required" label is rendered as a badge rather than as plain text at the top of the task. This is intended in order to be aligned with how other tasks emphasizes these small notices, like the classic editor deprecation notice: ![image](https://user-images.githubusercontent.com/1233880/98961771-3b4f0d00-2506-11eb-825b-2fcd97cef7bb.png)

#### Testing instructions

Note: You can use your Store Admin to modify the expiry date for individual plans.

- Apply D47225-code and sandbox the API.
- With Simple, Atomic, and Jetpack sites, check that:
  - No renew/reactivate tasks are shown for Free plans.
  - A renew task is shown for plans expiring in the next 30 days.
  - A reactivate task is shown for plans already expired.
  - For plan owners, make sure the primary button redirects to checkout.
  - For admins who are not the plan owner, make sure there is no primary button and they can dismiss the task only.
  - Non-admins shouldn't see this task (they cannot access My Home).

Fixes https://github.com/Automattic/wp-calypso/issues/43496
